Libraries/LibWeb/CSS/InvalidationSet.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2025, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/FlyString.h>
+#include <AK/Format.h>
+#include <AK/Forward.h>
+#include <AK/HashTable.h>
+#include <AK/Traits.h>
+
+namespace Web::CSS {
+
+class InvalidationSet {
+public:
+    struct Property {
+        enum class Type : u8 {
+            InvalidateSelf,
+            InvalidateWholeSubtree,
+            Class,
+            Id,
+            TagName,
+            Attribute,
+        };
+
+        Type type;
+        FlyString name {};
+
+        bool operator==(Property const& other) const = default;
+    };
+
+    void include_all_from(InvalidationSet const& other);
+
+    bool needs_invalidate_self() const { return m_properties.contains({ Property::Type::InvalidateSelf }); }
+    void set_needs_invalidate_self() { m_properties.set({ Property::Type::InvalidateSelf }); }
+
+    bool needs_invalidate_whole_subtree() const { return m_properties.contains({ Property::Type::InvalidateWholeSubtree }); }
+    void set_needs_invalidate_whole_subtree() { m_properties.set({ Property::Type::InvalidateWholeSubtree }); }
```

Comment:
Would it make more sense for these to be two `bool` members instead of entries in the property hash map?

Reply:
sure, I changed it to be two bools. I think we are going to end up replacing hash table with a vector for each property type because it seems wasteful to allocate a hash table for one id/class/attribute name, which is very often the case.